ui/src/app.js

```diff
@@ -587,6 +660,35 @@ $(document).ready(function () {
     buttonPressed('submitPromoCode', { input: code })
   })
 
+  const submitTextRequirementButton = document.getElementById('submit-text-requirement')
+  const nextFieldTextRequirementButton = document.getElementById('next-text-requirement')
+  const previousFieldTextRequirementButton = document.getElementById('previous-text-requirement')
+  touchEvent(submitTextRequirementButton, function () {
+    customRequirementTextKeyboard.deactivate.bind(customRequirementTextKeyboard)
+    var text = `${$('.text-input-field-1').data('content')} ${$('.text-input-field-2').data('content') || ''}`
+    buttonPressed('customInfoRequestSubmit', text)
+    $('.text-input-field-1').removeClass('faded').data('content', '').val('')
+    $('.text-input-field-2').addClass('faded').data('content', '').val('')
+    customRequirementTextKeyboard.setInputBox('.text-input-field-1')
+  })
+  touchEvent(nextFieldTextRequirementButton, function() {
+    $('.text-input-field-1').addClass('faded')
+    $('.text-input-field-2').removeClass('faded')
+    $('#next-text-requirement').hide()
+    $('#previous-text-requirement').show()
+    $('#submit-text-requirement').show()
+    // changing input box changes buttons where validation works on
+    customRequirementTextKeyboard.setInputBox('.text-input-field-2', ['#submit-text-requirement'])
+  })
+  touchEvent(previousFieldTextRequirementButton, function() {
+    $('.text-input-field-1').removeClass('faded')
+    $('.text-input-field-2').addClass('faded')
+    $('#next-text-requirement').show()
+    $('#previous-text-requirement').hide()
+    $('#submit-text-requirement').hide()
+    customRequirementTextKeyboard.setInputBox('.text-input-field-1', ['#next-text-requirement'])
+  })
```

Comment:
I'm not sure on how it is being done in other `touchEvent` calls, but wouldn't it be better for the callback to be encapsulated in their own functions? Mostly for readability purposes, as having the callback explicitly declared, it's easier to read `touchEvent(previousFieldTextRequirementButton, callbackFunction)`

Reply:
I don't think so in this case, it will just make you jump around the code in a file that's already huge...
plus, the other touchEvents also are like this